pandas/tests/groupby/test_function.py

```diff
@@ -26,6 +26,18 @@
 from pandas.util import _test_decorators as td
 
 
+@pytest.fixture(
+    params=[np.int32, np.int64, np.float32, np.float64],
+    ids=["np.int32", "np.int64", "np.float32", "np.float64"],
+)
+def numpy_dtypes(request):
```

Comment:
can you name this something else, maybe numpy_dtypes_for_minmax; also you *can* include in the return the iinfo or finfo result (e.g. you would return a tuple); i find this a bit more obvious than doing it in the test function.

Reply:
Updated based on comments and green

Reply:
not a fan of returning tuples and unpacking in tests in scenarios where composable fixtures could be used instead and could have more utility. However, naming the fixture `numpy_dtypes_for_minmax` somewhat limits the reuse of composible fixtures.

```diff
diff --git a/pandas/tests/groupby/test_function.py b/pandas/tests/groupby/test_function.py
index 6205dfb87..348a6c189 100644
--- a/pandas/tests/groupby/test_function.py
+++ b/pandas/tests/groupby/test_function.py
@@ -35,15 +35,31 @@ def numpy_dtypes_for_minmax(request):
     Fixture of numpy dtypes with min and max values used for testing
     cummin and cummax
     """
-    dtype = request.param
-    min_val = (
-        np.iinfo(dtype).min if np.dtype(dtype).kind == "i" else np.finfo(dtype).min
-    )
-    max_val = (
-        np.iinfo(dtype).max if np.dtype(dtype).kind == "i" else np.finfo(dtype).max
+    return request.param
+
+
+@pytest.fixture()
+def dtype_min(numpy_dtypes_for_minmax):
+    """
+    Fixture to return minimum value of dtype.
+    """
+    return (
+        np.iinfo(numpy_dtypes_for_minmax).min
+        if np.dtype(numpy_dtypes_for_minmax).kind == "i"
+        else np.finfo(numpy_dtypes_for_minmax).min
     )
 
-    return (dtype, min_val, max_val)
+
+@pytest.fixture()
+def dtype_max(numpy_dtypes_for_minmax):
+    """
+    Fixture to return maximum value of dtype.
+    """
+    return (
+        np.iinfo(numpy_dtypes_for_minmax).max
+        if np.dtype(numpy_dtypes_for_minmax).kind == "i"
+        else np.finfo(numpy_dtypes_for_minmax).max
+    )
 
 
 @pytest.mark.parametrize("agg_func", ["any", "all"])
@@ -704,9 +720,8 @@ def test_numpy_compat(func):
     reason="https://github.com/pandas-dev/pandas/issues/31992",
     strict=False,
 )
-def test_cummin(numpy_dtypes_for_minmax):
-    dtype = numpy_dtypes_for_minmax[0]
-    min_val = numpy_dtypes_for_minmax[1]
+def test_cummin(numpy_dtypes_for_minmax, dtype_min):
+    dtype = numpy_dtypes_for_minmax
 
     # GH 15048
     base_df = pd.DataFrame(
@@ -723,8 +738,8 @@ def test_cummin(numpy_dtypes_for_minmax):
     tm.assert_frame_equal(result, expected)
 
     # Test w/ min value for dtype
-    df.loc[[2, 6], "B"] = min_val
-    expected.loc[[2, 3, 6, 7], "B"] = min_val
+    df.loc[[2, 6], "B"] = dtype_min
+    expected.loc[[2, 3, 6, 7], "B"] = dtype_min
     result = df.groupby("A").cummin()
     tm.assert_frame_equal(result, expected)
     expected = df.groupby("A").B.apply(lambda x: x.cummin()).to_frame()
@@ -772,9 +787,8 @@ def test_cummin_all_nan_column():
     reason="https://github.com/pandas-dev/pandas/issues/31992",
     strict=False,
 )
-def test_cummax(numpy_dtypes_for_minmax):
-    dtype = numpy_dtypes_for_minmax[0]
-    max_val = numpy_dtypes_for_minmax[2]
+def test_cummax(numpy_dtypes_for_minmax, dtype_max):
+    dtype = numpy_dtypes_for_minmax
 
     # GH 15048
     base_df = pd.DataFrame(
@@ -791,8 +805,8 @@ def test_cummax(numpy_dtypes_for_minmax):
     tm.assert_frame_equal(result, expected)
 
     # Test w/ max value for dtype
-    df.loc[[2, 6], "B"] = max_val
-    expected.loc[[2, 3, 6, 7], "B"] = max_val
+    df.loc[[2, 6], "B"] = dtype_max
+    expected.loc[[2, 3, 6, 7], "B"] = dtype_max
     result = df.groupby("A").cummax()
     tm.assert_frame_equal(result, expected)
     expected = df.groupby("A").B.apply(lambda x: x.cummax()).to_frame()
```

Reply:
If we take the min and max out of the fixture I'd just say compute inside the test, since the min and max are each only used in one test respectively, so there isn't much value in having them in a fixture anyways

Reply:
Sorry I wasn't clear. I'm not a fan but no need to change. see https://github.com/pandas-dev/pandas/pull/32027#discussion_r379842378